omegatrade-app/backend/app/controller/simulation.controller.js

```diff
@@ -0,0 +1,123 @@
+'use strict';
+const Simulation = require('../models/simulation.model')
+const Company = require('../models/company.model')
+const { v4: uuidv4 } = require('uuid');
+const { spannerNumericRandVal, spannerNumericRandValBetween, generateRandomValue } = require('../helpers/stockdata.helper');
+const fakeStockmarketgenerator = require('fake-stock-market-generator');
+const { Spanner } = require('@google-cloud/spanner');
+
+
+/**
+ * Function to list all simulations.
+ * 
+ * @method GET
+ */
+exports.getList = async function (req, res) {
+    try {
+        await Simulation.getAll(function (err, data) {
+            if (err) {
+               return res.json({ success: false, message: "something went wrong" });
+            }
+            if (data) {
+               return res.status(200).json({ success: true, data: data });
+            }
+        });
+    } catch (error) {
+        return res.status(500).json({ success: false, message: "something went wrong" });
+    }
+};
+
+/**
+ * Function to update simulation
+ * 
+ * @method PUT
+ */
+exports.updateSimulation = async function (req, res) {
+    const body = req.body;
+    if (body) {
+        await Simulation.updateById(body, function (err, data) {
+            if (err) {
+               return res.json({ success: false, message: "something went wrong" });
+            }
+            if (data) {
+               return res.status(200).json({ success: true, message: `Simulation ${(body.status == true) ? 'Started' : 'Stopped'}  sucessfully` });
+            }
+        });
+    } else {
+        return res.status(501).json({ success: false, message: "invalid data" });
+    }
+}
+
+/**
+ * Function to Delete simulation
+ * 
+ * @method DELETE
+ */
+exports.deleteSimulation = async function (req, res) {
+    let sId = req.params.sId;
+    if (sId) {
+        await Simulation.deleteById(sId, function (err, data) {
+            if (err) { return res.json({ success: false, message: "something went wrong" }); }
+            if (data) { return res.status(200).json({ success: true, message: `deleted sucessfully` }); }
+        });
+    } else {
+       return res.status(501).json({ success: false, message: "Invalid data" });
+    }
+};
+
+exports.startSimulation = async function (req, res) {
+    try {
+        const body = req.body;
+        const [result] = await Company.findById(body.companyId);
+        if (result && result.length > 0) {
+            const company = result[0];
+            const interval = body.timeInterval * 1000;
+            const stock = fakeStockmarketgenerator.generateStockData(body.data).priceData;
+            const sId = await Simulation.create(company.companyId);
+            if (sId) {
```

Comment:
this if statement is not required, `simulation.create` function will return `sId` for sure. in case of any error, it will be caught and return the error.